### ISSUE TITLE
Could not copy text in pdf when using font-family in ubuntu system

### ISSUE THREAD

**Issue by sonicx991** [wkhtmltopdf/wkhtmltopdf#4890]
if the html contains <span style="font-family:'Arial'">placeholder1</span>, the test of the output pdf could not be copied
in ubuntu 18.04 system(or other linux,docker etc).works fine in win10 system.And if use the default font(no more font-family).It could be copied as well.

**Comment by sonicx991** [comment]
@font-face {
            font-family: "Times New Roman";
            src: url("/usr/share/fonts/truetype/Times_New_Roman.ttf")
        }

        @font-face {
            font-family: "webdings";
            src: url("/usr/share/fonts/truetype/webdings.ttf")
        }

        @font-face {
            font-family: "Wingdings";
            src: url("/usr/share/fonts/truetype/Wingdings_2.TTF")
        }

Such kind of font-face has been used,and I have copied these fonts to  /usr/share/fonts/truetype/

**Comment by sonicx991** [comment]
if the html contains placeholder1->if the html contains "<span style="font-family:'楷体'">"

**Comment by PhilterPaper** [comment]
Does the Arial text display OK, and you only can't copy it into a buffer? And what do you mean by "the test of the output pdf"? Did you mean "the **rest** of the output pdf"? Can non-Arial text be copied, and just the Arial has problems?

Take a look at where your Arial font is coming from (assuming you _can_ see it in the document). Is it some oddball format such as .WOFF? I've heard of problems where such fonts are rendered as _graphics_ collections and not text, and cannot be copy-pasted as a result.

**Comment by sonicx991** [comment]
Yes.The Arial text display ok,but can't be copied."the test of the output pdf" means the pdf output by wkhtmltopdf[for test].The arial font is a ttf.I list the 1.html 1.pdf and arial font below.So you can see all the Info

[test5.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/5608511/test5.zip)

**Comment by PhilterPaper** [comment]
I unpacked your 1.pdf (with PDFtk) and saw why you couldn't select and copy the "Arial" text -- as I suspected, it was pure graphics (no text write commands).

I was unable to reproduce the problem on my Windows 10 box. I tried as the font-face src both my standard Arial font (/WINDOWS/Font/arial.ttf) and the one you sent. The two font files are identical, so it doesn't look like your Arial font got damaged somewhere. The PDF I produced (wkHTMLtoPDF 0.12.5 with patched qt) displays "Arial" text as selectable and can be copied. The stream is text output of 5 glyphs. I _do_ see that the "Arial" text is in a slightly different position when I created it, than in your 1.pdf -- can you double check that your 1.pdf matches your 1.html you sent me? I also see that the "cm" global page transformation matrix is slightly different.

1U.pdf is your original, uncompressed. 2U.pdf is from when I ran wkHTMLtoPDF against 1.html
[1U.pdf](https://github.com/wkhtmltopdf/wkhtmltopdf/files/5609688/1U.pdf)
[2U.pdf](https://github.com/wkhtmltopdf/wkhtmltopdf/files/5609691/2U.pdf)

**Comment by sonicx991** [comment]
yeah.I said before it works fine in win10,
and happenes in ubuntu 18.04.maybe in ubuntu,needs a xvfb_run command.or other reasons

**Comment by sonicx991** [comment]
By the way ,when using margin-left mm param,the results between windows and ubuntu are different.For ubuntu,the results seem little obvious,you have to increase the value to see the margins

**Comment by PhilterPaper** [comment]
Since your run on Ubuntu produced graphics draws instead of text output, I would concentrate on the Arial font file. You were definitely using a .ttf file copied over from Windows? I ask because I've heard of this behavior when using formats such as WOFF. If you were definitely using TTF, then it must be a Ubuntu/Linux glitch.

**Comment by sonicx991** [comment]
yeah I'm using ttf.so it seems no solution right now?

**Comment by PhilterPaper** [comment]
> yeah I'm using ttf.so it seems no solution right now?

I don't have anything. Sorry.
